src/main/java/com/future/function/qa/data/BaseData.java

```diff
@@ -1,27 +1,73 @@
 package com.future.function.qa.data;
 
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.future.function.qa.model.response.base.BaseResponse;
 import com.future.function.qa.model.response.base.DataResponse;
+import com.future.function.qa.model.response.base.ErrorResponse;
+import com.future.function.qa.model.response.base.PagingResponse;
 import io.restassured.http.Cookie;
 import io.restassured.response.Response;
 import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
 
 @Data
+@Slf4j
 public abstract class BaseData {
 
-  protected BaseResponse baseResponse;
+  protected BaseResponse baseResponse = new BaseResponse();
 
   protected Cookie cookie;
 
+  protected ErrorResponse errorResponse = new ErrorResponse();
+
+  @Autowired
+  protected ObjectMapper mapper;
+
   protected int responseCode;
 
-  public static BaseResponse asBaseResponse(Response response) {
+  public <T> DataResponse<T> asDataResponse(Response response, TypeReference<DataResponse<T>> typeReference) {
+
+    try {
+      return mapper.readValue(response.print(), typeReference);
+    } catch (Exception e) {
+      return new DataResponse<>();
```

Comment:
can this actually throw exception?

Reply:
Yes, so the idea was to have only 1 setter method for retrieved response (`BaseResponse`, `ErrorResponse`, `DataResponse`, or `PagingResponse`), which would set all those types of responses, should the retrieved response could be parsed. If not, then default value will be used (in this case: empty constructor)